src/System.Net.Http/src/System/Net/Http/SocketsHttpHandler/HPack/HPackDecoder.cs

```diff
@@ -374,10 +391,10 @@ public void CompleteDecode()
             }
         }
 
-        private void OnIndexedHeaderField(int index, HeaderCallback onHeader, object onHeaderState)
+        private void OnIndexedHeaderField(int index, IHttpHeadersHandler handler)
         {
             HeaderField header = GetHeader(index);
-            onHeader(onHeaderState, new ReadOnlySpan<byte>(header.Name), new ReadOnlySpan<byte>(header.Value));
+            handler?.OnHeader(header.Name, header.Value);
```

Comment:
Are we expecting `handler` to be `null` here?

Reply:
Well originally I didn't, but it can. https://github.com/dotnet/corefx/blob/master/src/System.Net.Http/src/System/Net/Http/SocketsHttpHandler/Http2Connection.cs#L338-L340. The stream can be null here, meaning the handler can be null as well. 

I believe this logic exists because you want to drain any frames after a stream went away.

Reply:
In that case, we need to make sure `handler` is assumed null elsewhere too, like in `ProcessHeaderValue()`.